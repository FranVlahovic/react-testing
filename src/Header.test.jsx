import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import Header from "./Header";
import { afterEach, expect } from "vitest";

afterEach(cleanup);

it("renders", () => {
    const { asFragment } = render(<Header text="hello" />);
    expect(asFragment()).toMatchSnapshot();
});

it("inserts text in h1", () => {
    const { getByTestId, getByText } = render(<Header text="hello two" />);

    expect(getByTestId('h1tag')).toHaveTextContent("hello two");
    expect(getByText("hello two")).toHaveClass("fancy-h1");
})