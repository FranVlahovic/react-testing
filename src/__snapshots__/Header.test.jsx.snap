// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders 1`] = `
<DocumentFragment>
  <header>
    <h1
      class="fancy-h1"
      data-testid="h1tag"
    >
      hello
    </h1>
  </header>
</DocumentFragment>
`;
